### PR TITLE
[#3038]Handle long option results

### PR DESCRIPTION
**Pull request by stellanl**
#### Before the PR (what is the issue or what needed to be done)
Long/JSON results are not handled when looking up summary data.
#### The solution
Take apart JSON results and handle it without a crash if it still is too long.
Also moved the task-submitting code to DataProcessorRestServelet, where the task will be handled.
## Checklist
* [ ] Connect the issue
* [ ] Test plan
* [ ] Copyright header
* [ ] Code formatting
* [ ] Documentation

### CODE REVIEW

File: GAE/src/org/waterforpeople/mapping/analytics/dao/SurveyQuestionSummaryDao.java [hunk 1]
```diff
@@ -61,27 +64,17 @@ public SurveyQuestionSummaryDao() {
...
-        String answerText = answer.getValue();
-        String[] answers;
-        if (answerText != null && answerText.contains("|")) {
-            answers = answerText.split("\\|");
-        } else {
-            answers = new String[] {
-                    answerText
-            };
-        }
+        String[]answers = DataUtils.optionResponsesTextArray(answer.getValue()); //JSON or |
```
Comment (muloem) [inline GAE/src/org/waterforpeople/mapping/analytics/dao/SurveyQuestionSummaryDao.java]: formatting looks weird... 🙁

File: GAE/src/org/waterforpeople/mapping/analytics/dao/SurveyQuestionSummaryDao.java [hunk 2]
```diff
@@ -124,6 +117,11 @@ public static synchronized void incrementCount(QuestionAnswerStore answer,
      */
     @SuppressWarnings("unchecked")
     public List<SurveyQuestionSummary> listByResponse(String questionId, String questionResponse) {
+        //Reject too-long key; TODO truncate instead?
```
Comment (muloem) [inline GAE/src/org/waterforpeople/mapping/analytics/dao/SurveyQuestionSummaryDao.java]: I suggest lets go with the truncate option on this one.

File: GAE/src/org/waterforpeople/mapping/analytics/dao/SurveyQuestionSummaryDao.java (hunk 1 above)
Comment (stellanl) [reply GAE/src/org/waterforpeople/mapping/analytics/dao/SurveyQuestionSummaryDao.java]: Fixed.

File: GAE/src/org/waterforpeople/mapping/analytics/dao/SurveyQuestionSummaryDao.java (hunk 2 above)
Comment (stellanl) [reply GAE/src/org/waterforpeople/mapping/analytics/dao/SurveyQuestionSummaryDao.java]: Ok.

File: GAE/src/org/waterforpeople/mapping/analytics/dao/SurveyQuestionSummaryDao.java [hunk 3]
```diff
@@ -124,10 +121,16 @@ public static synchronized void incrementCount(QuestionAnswerStore answer,
...
+        String answer = questionResponse;
+        if (answer.length() > Constants.MAX_LENGTH) {
```
Comment (muloem) [inline GAE/src/org/waterforpeople/mapping/analytics/dao/SurveyQuestionSummaryDao.java]: Just like previously do you want to check whether answer is null?

File: GAE/src/org/waterforpeople/mapping/analytics/dao/SurveyQuestionSummaryDao.java (hunk 2 above)
Comment (muloem) [reply GAE/src/org/waterforpeople/mapping/analytics/dao/SurveyQuestionSummaryDao.java]: Can as well clear out the comment

File: GAE/src/org/waterforpeople/mapping/analytics/dao/SurveyQuestionSummaryDao.java (hunk 3 above)
Comment (stellanl) [reply GAE/src/org/waterforpeople/mapping/analytics/dao/SurveyQuestionSummaryDao.java]: ok

File: GAE/src/org/waterforpeople/mapping/analytics/dao/SurveyQuestionSummaryDao.java (hunk 2 above)
Comment (stellanl) [reply GAE/src/org/waterforpeople/mapping/analytics/dao/SurveyQuestionSummaryDao.java]: done

**Pull request merged**